packages/atomic/src/components/atomic-search-box/atomic-search-box.tsx

```diff
@@ -99,6 +119,25 @@ export class AtomicSearchBox {
     this.pendingSuggestionEvents = [];
   }
 
+  public componentDidUpdate() {
+    if (!('redirectTo' in this.searchBoxState)) {
+      return;
+    }
+
+    const {redirectTo, value, analytics} = this.searchBoxState;
+
+    if (redirectTo) {
+      const data = {value, analytics};
+      const storage = new SafeStorage();
+      storage.setItem(
```

Comment:
Use `setJSON`, and let the storage handle the `JSON.stringify` (because any `JSON.stringify` can throw).